base/linalg/triangular.jl

```diff
@@ -1297,72 +1318,85 @@ for t in (UpperTriangular, UnitUpperTriangular, LowerTriangular, UnitLowerTriang
     end
 end
 
-for f in (:*, :Ac_mul_B, :At_mul_B, :\, :Ac_ldiv_B, :At_ldiv_B)
+# for f in (:*, :Ac_mul_B, :At_mul_B, :\, :Ac_ldiv_B, :At_ldiv_B)
+#     @eval begin
+#         ($f)(A::AbstractTriangular, B::AbstractTriangular) = ($f)(A, full(B))
+#     end
+# end
+for f in (:*, :Ac_mul_B, :At_mul_B)
     @eval begin
         ($f)(A::AbstractTriangular, B::AbstractTriangular) = ($f)(A, full(B))
     end
 end
-for f in (:A_mul_Bc, :A_mul_Bt, :Ac_mul_Bc, :At_mul_Bt, :/, :A_rdiv_Bc, :A_rdiv_Bt)
+# for f in (:A_mul_Bc, :A_mul_Bt, :Ac_mul_Bc, :At_mul_Bt, :/, :A_rdiv_Bc, :A_rdiv_Bt)
+#     @eval begin
+#         ($f)(A::AbstractTriangular, B::AbstractTriangular) = ($f)(full(A), B)
+#     end
+# end
+for f in (:A_mul_Bc, :A_mul_Bt, :Ac_mul_Bc, :At_mul_Bt)
     @eval begin
         ($f)(A::AbstractTriangular, B::AbstractTriangular) = ($f)(full(A), B)
     end
 end
 
 ## The general promotion methods
+for mat in (:AbstractVector, AbstractMatrix)
```

Comment:
It was necessary to split the union to avoid ambiguity warnings but now that https://github.com/JuliaLang/julia/pull/16125 is merged, I'll revisit this and see if it can be simplified.